### DISCUSSION TITLE
Aggregate content based on tag and content block

### DISCUSSION THREAD

**Discussion by vchryssos** [zim-desktop-wiki/zim-desktop-wiki#1818] [Ideas]
Hi there, 
Thank you so much for a great software! I really appreciate your effort and the openness in developing it. 

I am a heavy journal user, adding notes from my everyday work and interactions. Some cases span multiple days, so I'm keeping notes in various pages regarding the specific cases. 

It would be great if there was some way to aggregate all this content in a single page. 

The way I imagine this working is: 
* use a specific tag to mark some content
* enclose this content in a start/end block, e.g. 
tag:homedesign --
content here
--
* upon request, dynamically search all pages and aggregate this content from every journal page that contains the above blocks
* sort content by date, adding the date for each

The above functionality allows one to easily see the history of any specific case, while at the same time maintaining the notes in each day's journal.

**Comment by introt** [comment]
Hello there!

In the parallel wikiality of TiddlyWiki, this can be achieved by _transcluding_ Tiddlers (tiny (sub)pages) into other Tiddlers using the TranscludeWidget. It's really powerful and the number one feature from TiddlyWiki that I miss in Zim. Give it a try while you wait!

If you're in a hurry and a UNIX user with GNU AWK available, I made you a Custom Tool which should mostly cover your needs: https://github.com/jaap-karssenberg/zim-wiki/wiki/tag-block-aggregator-custom-tool

A future plugin implementation would probably use the upcoming embedded pageviews to allow you to edit the source pages' blocks without leaving the destination page. 

Related issues:
* related pages by matching tags: #39
* embedded pageview: block quotes, admonion blocks: #41,  #42
* "embed the content of the links [to other pages] into the note" #1373

**Comment by vchryssos** [comment]
I'm speechless @introt! Thnx for your time and effort. I will give it a try.

**Comment by introt** [comment]
No problem my friend; I don't get too many chances to flex my UNIX muscles and was planning to craft a similar tool anyway :)

If you encounter any issues, please report them [here](https://github.com/introt/zim-customtools/issues). Thanks!

**Reply by Lost-Projects** [reply]
I think this is a script invoked by a command. Under "usage" can you specify a path where the script should be placed for ignorant people like me? I can't see this step in the custom tool help page.
I'm using Linux mint - maybe Awk is included?

Does this update automatically or does the tool need to be manually triggered for each instance? Guessing the latter.
